frontend/app/components/redesign/components/LinkTagGenerator.tsx

```diff
@@ -0,0 +1,146 @@
+import { useState } from 'react'
+import { cx } from 'class-variance-authority'
+import { Heading5 } from '../Typography'
+import { InputField, ToolsPrimaryButton } from './index'
+import { SVGCopyIcon, SVGCheckIcon } from '~/assets/svg'
+
+export const LinkTagGenerator = () => {
+  const [pointerInput, setPointerInput] = useState('')
+  const [linkTag, setParsedLinkTag] = useState('')
+  const [invalidUrl, setInvalidUrl] = useState(false)
+  const [showCodeBox, setShowCodeBox] = useState(false)
+  const [isCopied, setIsCopied] = useState(false)
+
+  const isValidPointer = (input: string) => {
+    try {
+      let urlString = input.trim()
+      if (input.charAt(0) === '$') {
+        urlString = input.replace('$', 'https://')
+      }
+
+      const url = new URL(urlString)
+
+      if (url.pathname === '/') {
+        return `${url.origin}.well-known/pay`
+      }
+
+      return url.origin + url.pathname
+    } catch {
+      setInvalidUrl(true)
+    }
+  }
+
+  const parsePointer = (pointer: string) => {
+    return pointer
+      .replace(/&/g, '&amp;')
+      .replace(/</g, '&lt;')
+      .replace(/>/g, '&gt;')
+      .replace(/"/g, '&quot;')
+      .replace(/'/g, '&apos;')
+  }
+
+  const handleCopyClick = async () => {
+    try {
+      await navigator.clipboard.writeText(
+        `<link rel="monetization" href="${linkTag}" />`
+      )
+      setIsCopied(true)
+      setTimeout(() => setIsCopied(false), 2000)
```

Comment:
Might want to have this `setTimeout` in a `useEffect` for best UX. Otherwise, an older timeout can reset the `setIsCopied` if user clicks quickly.